python/pyarrow/__init__.py

```diff
@@ -44,13 +44,6 @@ def parse_git(root, **kwargs):
 
 import pyarrow.compat as compat
 
-# Workaround for https://issues.apache.org/jira/browse/ARROW-2657
```

Comment:
I'm +0 on removing this. If it makes people's lives better and removing it would make it impossible to run many workloads, happy to keep it. But there is also value in making clear that this cannot be made to work reliably, so I'm removing it unless there are objections.